shared/login/join-or-login/container.tsx

```diff
@@ -16,24 +18,39 @@ const mapStateToProps = state => {
     bannerMessage = `${state.devices.justRevokedSelf} was revoked successfully`
   }
 
-  return {bannerMessage}
+  return {
+    bannerMessage,
+    isOnline: state.login.isOnline,
+  }
 }
 
 const mapDispatchToProps = (dispatch, {navigateAppend}: OwnProps) => ({
+  _checkIsOnline: () => dispatch(LoginGen.createLoadIsOnline()),
   _onFeedback: () => dispatch(navigateAppend(['feedback'])),
   onLogin: () => dispatch(ProvisionGen.createStartProvision()),
   onSignup: () => dispatch(SignupGen.createRequestAutoInvite()),
+  showProxySettings: () => dispatch(navigateAppend(['proxySettingsModal'])),
 })
 
 const mergeProps = (stateProps, dispatchProps, ownProps: OwnProps) => ({
   bannerMessage: stateProps.bannerMessage,
+  checkIsOnline: dispatchProps._checkIsOnline,
+  isOnline: stateProps.isOnline,
   onFeedback: isMobile ? dispatchProps._onFeedback : null,
   onLogin: dispatchProps.onLogin,
   onSignup: dispatchProps.onSignup,
+  showProxySettings: dispatchProps.showProxySettings,
 })
 
-export default connect(
-  mapStateToProps,
-  mapDispatchToProps,
-  mergeProps
+export default compose(
+  connect(
+    mapStateToProps,
+    mapDispatchToProps,
+    mergeProps
+  ),
+  lifecycle({
```

Comment:
We're trying to deprecate using compose() and lifecycle(), because they're untyped.. looks like this goes through to a class component so can you just use that component's componentDidMount instead?

You could also consider using React Hooks useEffect() to keep that component as a function instead of class.

Reply:
Done, moved it over to that component's lifecycle methods.